src/ucp/wireup/wireup_ep.c

```diff
@@ -330,13 +330,30 @@ static ucs_status_t ucp_wireup_ep_flush(uct_ep_h uct_ep, unsigned flags,
     return UCS_ERR_NO_RESOURCE;
 }
 
+static ucs_status_t ucp_wireup_ep_check(uct_ep_h uct_ep, unsigned flags,
+                                        uct_completion_t *comp)
+{
+    ucp_wireup_ep_t *wireup_ep = ucp_wireup_ep(uct_ep);
+    ucp_worker_h worker        = wireup_ep->super.ucp_ep->worker;
+
+    if (wireup_ep->flags & UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED) {
+        return uct_ep_check(wireup_ep->super.uct_ep, flags, comp);
+    } else if ((wireup_ep->aux_ep != NULL) &&
```

Comment:
can remove 'else'
BTW, why need to check UCT_IFACE_FLAG_EP_CHECK flag? i would expect UCP keepalive mechanism to check this already

Reply:
removed
The check EP_CHECK flag is needed, since UCP KA mechanism checks it only for iface which `wireup_ep->super.uct_ep` belongs to (e.g. next_ep - RC, aux_ep - UD. but UD doesn't have EP_CHECK support)

Reply:
so we need to make sure only transports which have CHECK support are used as aux_ep

Reply:
then UD couldn't be used as aux_ep

Reply:
ok, is it a problem?
BTW need to make sure UCP would really send keepalive on the ep used for wireup messages, and that ep keepalive list is updated when ep configuration changes

Reply:
done, now doing EP check over UCT if connected, and otherwise - over AUX EP